modules/subscriptions.php

```diff
@@ -194,15 +206,23 @@ function maybe_send_subscription_email( $new_status, $old_status, $post ) {
 
 		// Only emails posts from these categories
 		if ( ! empty( $only_these_categories ) && ! in_category( $only_these_categories, $post->ID ) ) {
-			update_post_meta( $post->ID, '_jetpack_dont_email_post_to_subs', 1 );
+			return false;
```

Comment:
returning false from these if-statements is actually a change in functionality. In the old version, the control flow would hit the bottom of this function where these metas could be overridden by `$_POST['_jetpack_dont_email_post_to_subs']`